src/App.js

```diff
@@ -1,11 +1,34 @@
-import logo from './logo.svg';
-import './App.css';
+import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
+import Header from './components/Header';
+import Categories from './components/Categories';
+import ShowBooks from './components/ShowBooks';
 
 function App() {
   return (
-    <div className="App">
-        
-    </div>
+    <Router>
+      <div className="App">
+        <Routes>
+          <Route
+            path="/"
+            element={(
+              <>
+                <Header />
+                <ShowBooks />
+              </>
+          )}
+          />
+          <Route
+            path="/categories"
+            element={(
+              <>
+                <Header />
+                <Categories />
+              </>
+          )}
+          />
+        </Routes>
+      </div>
```

Comment:
- [ Optional] You did an amazing job to implement react routers, but you still can improve the way you did it.  try to just put Header component at the top of your Routes, so that you can only use it once. this will make your codes more clean and efficient. check for [more ](https://www.youtube.com/watch?v=ghxVYER-GCY&ab_channel=GhostTogether)

Reply:
Thank you will sure do @ishimwezachee